dom.go

```diff
@@ -169,8 +173,14 @@ func (h *HTML) Restore(old ComponentOrHTML) {
 	}
 
 	if prev, ok := old.(*HTML); ok && prev != nil {
-		h.restoreHTML(prev)
-		return
+		if h.text != "" && prev.text != "" {
+			h.restoreText(prev)
+			return
+		}
+		if h.tag != "" && prev.tag != "" && h.tag == prev.tag {
+			h.restoreHTML(prev)
+			return
+		}
```

Comment:
Also this would be written more straightforward like this I think:

```
if h.text != "" && prev.text != "" {
	// Two text nodes.
	h.restoreText(prev)
	return
} else if h.tag == prev.tag {
	// Two HTML elements of the same tag.
	h.restoreHTML(prev)
	return
}
```

I'll make this change after merging.